src/LiveComponent/src/LiveComponentHydrator.php

```diff
@@ -493,14 +607,26 @@ private function dehydrateValue(mixed $value, LivePropMetadata $propMetadata, ob
             throw new \LogicException(\sprintf('Unable to dehydrate value of type "%s" for property "%s" on component "%s". Change this to a simpler type of an object that can be dehydrated. Or set the hydrateWith/dehydrateWith options in LiveProp or set "useSerializerForHydration: true" on the LiveProp to use the serializer.', get_debug_type($value), $propMetadata->getName(), $parentObject::class));
         }
 
-        if (!$propMetadata->getType() || $propMetadata->isBuiltIn()) {
-            throw new \LogicException(\sprintf('The "%s" property on component "%s" is missing its property-type. Add the "%s" type so the object can be hydrated later.', $propMetadata->getName(), $parentObject::class, $value::class));
-        }
+        // BC layer when "symfony/type-info" is not available
+        if ($propMetadata instanceof LegacyLivePropMetadata) {
+            if (!$propMetadata->getType() || $propMetadata->isBuiltIn()) {
+                throw new \LogicException(\sprintf('The "%s" property on component "%s" is missing its property-type. Add the "%s" type so the object can be hydrated later.', $propMetadata->getName(), $parentObject::class, $value::class));
+            }
+
+            // at this point, we have an object and can assume $propMetadata->getType()
+            // is set correctly (needed for hydration later)
 
-        // at this point, we have an object and can assume $propMetadata->getType()
-        // is set correctly (needed for hydration later)
+            return $this->dehydrateObjectValue($value, $propMetadata->getType(), $propMetadata->getFormat(), $parentObject);
+        } else {
+            foreach ($propMetadata->getType()?->traverse() ?? [] as $t) {
+                if ($t instanceof ObjectType) {
+                    return $this->dehydrateObjectValue($value, $t->getClassName(), $propMetadata->getFormat(), $parentObject);
+                }
+            }
+            dd($propMetadata);
```

Comment:
To remove

Reply:
Added to the mental list: automatically check for undesired functions call (dd, dump, ...)